internal/bminventory/inventory.go

```diff
@@ -5412,3 +5412,20 @@ func (b *bareMetalInventory) UnbindHost(ctx context.Context, params installer.Un
 
 	return installer.NewUnbindHostOK().WithPayload(&host.Host)
 }
+
+func (b *bareMetalInventory) V2ListHosts(ctx context.Context, params installer.V2ListHostsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	var hosts []*models.Host
+	if err := b.db.Find(&hosts, "infra_env_id = ?", params.InfraEnvID).Error; err != nil {
```

Comment:
may be we should add a list function to the internal/common.db. This way all the DB access/queries will be in the same file. What do you think?